wp-seo-main.php

```diff
@@ -27,6 +27,69 @@
 	define( 'WPSEO_CSSJS_SUFFIX', ( ( defined( 'SCRIPT_DEBUG' ) && true === SCRIPT_DEBUG ) ? '' : '.min' ) );
 }
 
+global $pagenow;
+if ( is_admin() && $pagenow === 'plugins.php' ) {
+	add_action( 'admin_init', 'yoast_seo_toggle_register_deactivate_listener' );
+}
+
+/**
+ * Listener for Yoast SEO to toggle Premium or Free when one of them is active.
+ */
+function yoast_seo_toggle_register_deactivate_listener() {
+	if ( ( $listener_plugin = filter_input( INPUT_GET, 'plugin' ) ) && ! empty( $listener_plugin ) ) {
```

Comment:
I would put the content of this if in a function like `yoast_seo_toggle_deactivate_redirect` or something like that.